course-outline.md

```diff
@@ -0,0 +1,105 @@
+
+# Suggested Learning Objectives / Outline of Each Section
+
+The overall idea of this course can be articulated as introducing students to enough tooling knowledge that they are comfortable creating and maintaining an online coding/technical resource, such as a JupyterBook. This will cover the skills they need to work with any type of literate coding document under version control:
+
+- Use the shell to move around on their computer
+- Use git and github to version control their project
+- Have a good project organization
+- Install jupytet-book in an isolated virutal env
+- Create reproducible notebook documents and version control them
+- Build and upload a Jupyter Book
+
+## 1. Intro
+
+1. Introduction to the toolbox, give an overview of what each tool does and where it fits in the toolchain
+2. Intro to VS Code
+    - I think this makes sense because it lets us use vscode for markdown documents, merge conflicts, and graphical git views in the following chapters. It can also be used to open files from terminal via `code <filename>`, which jupyterlab can't.
+        - I am hesitant to fully teach the graphical git interface from vscode and jupyterlab, but we can discuss it.
+3. Create minimal reproducible code examples when asking for help.
+    - [Lec8 521 first half](https://pages.github.ubc.ca/mds-2021-22/DSCI_521_platforms-dsci_students/materials/lectures/8-asking-effectively-and-virtual-environments.html)
+
+- Assignment 1 will be [Installing things](https://ubc-mds.github.io/resources_pages/installation_instructions/)
+    - We could either give this as part of lab0 only, or show how to do some of it during lecture as well.
+    - [Simpler installation version in dsci100 textbook](https://datasciencebook.ca/move-to-your-own-machine.html)
+    - They can upload a screenshot from the terminal of running our script as proof that it worked.
+
+## 2. Shell
```

Comment:
Make sure to test all 521 exercises and slides in JupyterLab terminal on all OSes

Reply:
Also add "How to use the terminal in jlab"

Reply:
Include material from Florencias PR in this section once we decide where it goes